src/test/java/org/elasticsearch/test/store/MockDirectoryHelper.java

```diff
@@ -122,11 +127,26 @@ public FsDirectoryService randomDirectorService(IndexStore indexStore) {
         private final boolean crash;
         private volatile RuntimeException closeException;
         private final Object lock = new Object();
+        private final Set<String> superUnSyncedFiles;
+        private final Random superRandomState;
 
         public ElasticsearchMockDirectoryWrapper(Random random, Directory delegate, ESLogger logger, boolean crash) {
             super(random, delegate);
             this.crash = crash;
             this.logger = logger;
+
+            // TODO: remove this silliness on Lucene upgrade:
+            try {
+                Field field = MockDirectoryWrapper.class.getDeclaredField("unSyncedFiles");
+                field.setAccessible(true);
+                superUnSyncedFiles = (Set<String>) field.get(this);
+
+                field = MockDirectoryWrapper.class.getDeclaredField("randomState");
+                field.setAccessible(true);
+                superRandomState = (Random) field.get(this);
```

Comment:
> why is it necessary to use the MockDirectoryWrapper's Random instead of the Random passed in to the ElasticsearchMockDirectoryWrapper

Hmm good question ... I was just trying to match what MDW did before Lucene switched to MockFS.

But looking at MDW's reasoning / comment in its ctor: it's necessary that we "fork" the incoming Random to our own private Random because the MDW APIs are accessed from multiple threads, and Lucene's test-framework gets (justifiably) angry when it detects that you are sharing a single Random instance across threads...